File: Homeworks/计导作业4/Fib2.go
package main
import "fmt"
import "time"
func fibonacci(n int)int{
    if n<=1{
        return n
    }
    prev,curr:=0,1
    for i:=2;i<=n;i++{
        prev,curr=curr,prev+curr
    }
    return curr
}

func main(){
    start:=time.Now()
    for i:=1;i<=50;i++{
        result:=fibonacci(i)
        fmt.Printf("F(%d)=%d\n",i,result)
    }
    fmt.Printf("F(50)=%d\n",fibonacci(50))
    t:=time.Since(start)
    fmt.Printf("Time: %v\n",t)
}

File: Homeworks/计导作业4/Fib2.py
import time
def fibonacci(n):
    if n<=1:
        return n
    a,b=0,1
    for i in range(2, n + 1):
        a,b=b,a+b
    return b
start=time.time()
for i in range(1, 51):
    result=fibonacci(i)
    print(f"F({i:d})={result}\n", end="")
end=time.time()
print(f"F(50)={fibonacci(50)}")
print(f"Time:{(end-start):.4f}")
